### PR TITLE
Add new value value type for wide-column entities

### PR DESCRIPTION
**Pull request by ltamasi**
Test Plan:
`make check`

### CODE REVIEW

File: db/dbformat.h [hunk 1]
```diff
@@ -77,7 +79,7 @@ extern const ValueType kValueTypeForSeekForPrev;
 // (i.e. a type used in memtable skiplist and sst file datablock).
 inline bool IsValueType(ValueType t) {
   return t <= kTypeMerge || t == kTypeSingleDeletion || t == kTypeBlobIndex ||
-         kTypeDeletionWithTimestamp == t;
+         t == kTypeDeletionWithTimestamp || t == kTypeWideColumnEntity;
```
Comment (riversand963) [inline db/dbformat.h:82]: I feel `kTypeDeletionWithTimestamp == t` is safer because you cannot mistaken it with `t = kTypeDeletionWithTimestamp`.

**Review by riversand963 — APPROVED** [review]
Otherwise, LGTM, assuming this will be announced in the PR activating the feature, especially the data format compatibility.

**Comment by ltamasi** [comment]
> Otherwise, LGTM, assuming this will be announced in the PR activating the feature, especially the data format compatibility.

Right, this is just groundwork; will add a changelog entry in a later patch.

File: db/dbformat.h (hunk 1 above)
Comment (ltamasi) [reply db/dbformat.h:82]: Yeah, I just wasn't a fan of the inconsistency across value types :) Will change these to be the other way around.

**Pull request closed without merging**